### PR TITLE
Stop Kafka message processing when closing SocketIO connection

### PR DESCRIPTION
**Pull request by eduardogmisiuk**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)
The Kafka Messenger listener keeps processing messages even when the WebSocket connection has terminated.

* **What is the new behavior (if this is a feature change)?**
The messages are processed only when there is an active SocketIO connection.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
https://github.com/dojot/dojot/issues/1387

* **Other information**:

### CODE REVIEW

File: src/SocketIOHandler.ts [hunk 1]
```diff
@@ -137,6 +155,38 @@ class SocketIOHandler {
...
   }
 
+  /**
+   * Register a callback in Kafka Messenger events.
+   * @param subject Kafka subject
+   * @param event
+   * @param callback callback to be registered for Kafka Messenger
+   * @param token the token returned by the Socket
+   */
+  private registerCallback(subject: string, event: string, cb: (ten: string, data: any) => void, token: string): void {
```
Comment (rascaraficci) [inline src/SocketIOHandler.ts]: It seems that the handlers will be registered at the Kafka Messenger only when the first socket.io connection is established. But, it seems that the callbacks are removed when the socket.io connection that cause the registration is closed. What happens with other socket.io connections that are still opened?  Shouldn't the callback unregistered when the last socket.io connection is closed?

Comment (eduardogmisiuk) [reply src/SocketIOHandler.ts]: From what I tested, the scenarios for SocketIO connection opening that we have in the current Dojot version are:
1) Clicking in notifications
2) Clicking in a device to see its dashboard
3) Clicking in a attribute that is a geopoint

In 1) we have only one connection, that will end when we exit from the notifications.
In 3) we open a connection when we already have the connection 2) open, and the connection to the map will close when we leave the device's dashboard, so there will be no connections that need the handlers.

Comment (rascaraficci) [reply src/SocketIOHandler.ts]: That's right for one browser, but if a open multiple browsers?

Comment (eduardogmisiuk) [reply src/SocketIOHandler.ts]: Haven't thought about it. I will test and modify if necessary, thanks!

Comment (rascaraficci) [reply src/SocketIOHandler.ts]: I think you can implement some kind of reference count with few changes.

**Pull request merged**